### DISCUSSION TITLE
is the stm32h747xihx-pinctrl.dtsi generated during the build ?

### DISCUSSION THREAD

**Discussion by othmaneAZ** [zephyrproject-rtos/zephyr#30114] [General] [answered]
Hello everyone,
**(This is not a bug, I'm just trying to understand)**

* Zephyr Version 2.4.0
* Board : stm32h747i_disco_m7

Hello everyone, I'm trying to port a new board on Zephyr, this board is based on an STM32H7 MCU. To do so, I was taking a look on the STM32H747X discovery board device tree sources and includes. I found out the device tree **stm32h747i_disco_m7.dts**is including the **stm32h747xihx-pinctrl.dtsi** file and I couldn't find any reference of this **.dtsi** file nor on the include directory not on the internet. 
`/dts-v1/; `
`#include <st/h7/stm32h747Xi_m7.dtsi>`
`#include <st/h7/stm32h747xihx-pinctrl.dtsi>`
`#include "stm32h747i_disco.dtsi"`

I'm wondering if it is a generated file. If it is the case I would like to understand at which stage it is generated ?

Thank you,

**Comment by erwango** [comment, marked as answer]
@othmaneAZ -pinctrl.dtsi files are made available in [hal_stm32](https://github.com/zephyrproject-rtos/hal_stm32) module (under `dts/st`)
You might need to run `west update` on a recent master to see them showing up.

**Reply by othmaneAZ** [reply]
Thank you for your reply, I found them !
